extensions/shared/paid-blocks/components.jsx

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/components';
+
+export const PremiumIcon = ( { icon, name } ) =>{
+	const blockNames = name.split( '/' );
+	const iconProps = icon?.src ? { icon: icon.src } : { icon };
+	const cssClass = 'jetpack-premium-icon' + ( blockNames?.[ 1 ] ? ` jetpack-premium-${ blockNames[ 1 ] }-icon` : '' );
+
+	return (
+		<div className={ cssClass }>
```

Comment:
If we do this wrapping approach, this would need to be a `<span>` since it gets embedded within a `<button>` and `<button><div /></button>` is invalid HTML.

Reply:
👍 feel free to update.